File: src/transformers/features/protobuf/ProtobufCreateValidateDecodeTransformer.ts
import { ProtobufValidateDecodeProgrammer } from "../../../programmers/protobuf/ProtobufValidateDecodeProgrammer";

import { GenericTransformer } from "../../internal/GenericTransformer";

export namespace ProtobufCreateValidateDecodeTransformer {
  export const transform = GenericTransformer.factory(
    "protobuf.createValidateDecode",
  )(
    (project) => (modulo) =>
      ProtobufValidateDecodeProgrammer.write(project)(modulo),
  );
}

File: packages/transform/src/features/protobuf/ProtobufCreateValidateDecodeTransformer.go
package protobuf

import (
	coreprotobuf "github.com/samchon/typia/packages/core/src/programmers/protobuf"
	internal "github.com/samchon/typia/packages/transform/src/internal"
)

var ProtobufCreateValidateDecodeTransformer = protobufCreateValidateDecodeTransformerNamespace{}

type protobufCreateValidateDecodeTransformerNamespace struct{}

func (protobufCreateValidateDecodeTransformerNamespace) Transform(call internal.ITypiaCallExpression) string {
	return coreprotobuf.ProtobufValidateDecodeProgrammer.Write(call.TypeText)
}
